src/main/java/com/google/sps/servlets/EditRecipeServlet.java

```diff
@@ -68,16 +69,22 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
       }
 
       recipeEntity.setProperty("name", name);
-      recipeEntity.setProperty("tags", tags);
       recipeEntity.setProperty("description", description);
+      recipeEntity.setProperty("imageBlobKey", imageBlobKey);
       recipeEntity.setProperty("ingredients", ingredients);
       recipeEntity.setProperty("steps", steps);
-      recipeEntity.setProperty("imageBlobKey", (String) recipeEntity.getProperty("imageBlobKey"));
+      recipeEntity.setProperty("tags", tags);
+
+      if (privacy.equals("public")) {
+        recipeEntity.setProperty("published", true);
+      }
+      if (privacy.equals("private")) {
+        recipeEntity.setProperty("published", false);
+      }
```

Comment:
If privacy equals some other value besides public and private, can you default it to published = false? It seems like you should either still set the property, or return an error, rather than fail silently.

also nit, but can you make the second if an else if? These 2 statements should be mutually exclusive, so no need to check them both if the first is true.

Reply:
so after the check for public I'll just have an else that sets published false then